### Q1 and 2. Import pakcage and load the data

In [1]:
import turicreate as tc

In [2]:
sf = tc.SFrame('./data/people_wiki.sframe')

In [3]:
# display first few lines...
sf

Columns:
	URI	str
	name	str
	text	str

Rows: 59071

Data:
+-------------------------------+---------------------+
|              URI              |         name        |
+-------------------------------+---------------------+
| <http://dbpedia.org/resour... |    Digby Morrell    |
| <http://dbpedia.org/resour... |    Alfred J. Lewy   |
| <http://dbpedia.org/resour... |    Harpdog Brown    |
| <http://dbpedia.org/resour... | Franz Rottensteiner |
| <http://dbpedia.org/resour... |        G-Enka       |
| <http://dbpedia.org/resour... |    Sam Henderson    |
| <http://dbpedia.org/resour... |    Aaron LaCrate    |
| <http://dbpedia.org/resour... |   Trevor Ferguson   |
| <http://dbpedia.org/resour... |     Grant Nelson    |
| <http://dbpedia.org/resour... |     Cathy Caruth    |
+-------------------------------+---------------------+
+-------------------------------+
|              text             |
+-------------------------------+
| digby morrell born 10 octo... |
| alfred j lewy aka sandy le... |
| harpdog brown is a singer ... |
| franz rottensteiner born i... |
| henry krvits born 30 decem... |
| sam henderson born october... |
| aaron lacrate is an americ... |
| trevor ferguson aka john f... |
| grant nelson born 27 april... |
| cathy caruth born 1955 is ... |
+-------------------------------+
[59071 rows x 3 columns]
Note: Only the head of the SFrame is printed.
You can use print_rows(num_rows=m, num_columns=n) to print more rows and columns.

In [4]:
sf['text'].summary()


+------------------+-------+----------+
|       item       | value | is exact |
+------------------+-------+----------+
|      Length      | 59071 |   Yes    |
| # Missing Values |   0   |   Yes    |
| # unique values  | 59093 |    No    |
+------------------+-------+----------+

Most frequent items:
+-------------------------------+-------+
|             value             | count |
+-------------------------------+-------+
| dominick a dom flora born ... |   3   |
| robin macpherson born 1959... |   3   |
| aaron lacrate is an americ... |   2   |
| joerg steineck is a german... |   2   |
| christopher sorrentino bor... |   2   |
| elizabeth gunn is an ameri... |   2   |
| prof dr ulfdietrich reips ... |   2   |
| agnes baltsa greek a m bor... |   2   |
| katja mira herbers dutch p... |   2   |
| neil shaka hislop cm born ... |   2   |
+-------------------------------+-------+


### Q3 - Which name is in the last row?

In [5]:
sf.tail(1)

Columns:
	URI	str
	name	str
	text	str

Rows: 1

Data:
+-------------------------------+--------------+-------------------------------+
|              URI              |     name     |              text             |
+-------------------------------+--------------+-------------------------------+
| <http://dbpedia.org/resour... | Fawaz Damrah | fawaz mohammed damrah arab... |
+-------------------------------+--------------+-------------------------------+
[1 rows x 3 columns]

### Q4 - Read the text column for Harpdog Brown. He was honored with:

In [6]:
sf[(sf['name'] == "Harpdog Brown")]['text']   # sf['text'][2] 

dtype: str
Rows: ?
['harpdog brown is a singer and harmonica player who has been active in canadas blues scene since 1982 hailing from vancouver he crossed tens of thousands of miles playing club dates and festivals in canada the northwestern united states and germanyover the years he has issued seven cds in 1995 his home is where the harp is won the muddy award for the best nw blues release from the cascade blues association in portland oregon as well that year it was nominated for a canadian juno for the best bluesgospel recording teamed up with graham guest on piano his cd naturally was voted 1 canadian blues album of 2010 by the blind lemon surveybrown tours extensively with his guitarist j arthur edmonds performing their electric mid1950s chicago blues either as a duo or with the full band while he is home he juggles a few combos working many venues big and small he also leads the harpdog brown band which is a gutsy traditional chicago blues band in 2014 they released what it is c

Read the text column for Harpdog Brown. He was honored with:  
  - [ ] A gold harmonica to recognize his innovative playing style.
  - [ ] A Grammy award for his latest blues album.
  - [X] A lifetime membership in the Hamilton Blues Society.

### Q5. What is the name entry in the first row?

Sort the SFrame according to the text column, in ascending order

cf. [Sort an SFrame](https://apple.github.io/turicreate/docs/api/generated/turicreate.SFrame.sort.html)

In [7]:
sf.sort('text', ascending=True).head(3)

Columns:
	URI	str
	name	str
	text	str

Rows: 3

Data:
+-------------------------------+---------------------+
|              URI              |         name        |
+-------------------------------+---------------------+
| <http://dbpedia.org/resour... |     108 (artist)    |
| <http://dbpedia.org/resour... |        1.8.7        |
| <http://dbpedia.org/resour... | Diane Carlson Evans |
+-------------------------------+---------------------+
+-------------------------------+
|              text             |
+-------------------------------+
| 108 born 1978 is an italia... |
| 187 is a pseudonym for jor... |
| 1lt diane carlson evans an... |
+-------------------------------+
[3 rows x 3 columns]

- [ ] 8 Ball Aitken
- [X] 108 (artist) 
- [ ] Zygfryd Szo
- [ ] 007 James Bond
- [ ] Digby Morrell 